### DISCUSSION TITLE
Maya Reference Workflow

### DISCUSSION THREAD

**Discussion by womanyoyoyo-adsk** [Autodesk/maya-usd#2665] [Show and tell]
Here is a video that highlights the Maya Reference workflow released in MayaUSD 0.19.0: https://www.youtube.com/watch?v=XW9Pu2ep-kM 

This feature lets you work with Maya data alongside USD data until you are ready to cache the Maya data to USD. It was originally developed by Animal Logic's [AL_USDMaya plugin](https://github.com/AnimalLogic/AL_USDMaya). You will need Maya 2023.2 to see what is in the video. 

Link to documentation is [here](https://help.autodesk.com/view/MAYAUL/2023/ENU/?guid=GUID-EDC40BE3-2112-46D5-9E31-B620C08C8DDE). Note our limitation of working with mixed-data models, for example if you select the USD parent node of Maya data, the Maya data nested inside will not be a part of that selection. Also, when the USD parent of Maya data gets deactivated/deleted/switched, the Maya data is orphaned and still appears in the viewport. This causes the MayaReference prim's 'Maya Auto Edit' attribute to not function.

Thank you for being patient! We are very much looking forward to your feedback.

**Comment by brycegbrazen** [comment]
@womanyoyoyo-adsk Regarding the limitations mentioned above, are these anticipated to be temporary limitations that are actively being worked out? Or are these more permanent limitations?

**Reply by womanyoyoyo-adsk** [reply]
The Maya data being orphaned and still showing in the viewport we are currently working on. 

The mixed data model workflows we'd like to get feedback on. The Maya Reference comes in a prim, the MayaReference prim, and a user has to enable 'Edit As Maya Data' to turn it into Maya data. The Maya data then appears in the USD hierarchy. This state is what we call the mixed data model state and workflows include selecting an ancestor prim of the Maya data and running operations on this prim expecting the Maya data to also inherit these operations (framing, selecting, TRS, etc.). Would artists do this?

**Comment by ragtag** [comment]
Is there currently a way to cache normally referenced Maya data to USD, without having it be referenced under a USD prim?

Basically doing a similar workflow with what you would do when exporting animated and deforming Maya meshes to Alembic, but exporting them to USD instead.

**Reply by womanyoyoyo-adsk** [reply]
Yes. You can select the Maya Reference at the root and then File - Export Selected. Make sure you have the 'Files of Type' dropdown set to USD Export. For the 'Cache to USD' workflow I describe above, our cache is generated using this same export code. So by exporting your Maya data using File - Export Selected, you are creating a USD cache manually.
